python/pyspark/sql/pandas/conversion.py

```diff
@@ -103,10 +103,22 @@ def toPandas(self):
                     batches = self.toDF(*tmp_column_names)._collect_as_arrow()
                     if len(batches) > 0:
                         table = pyarrow.Table.from_batches(batches)
+                        del batches
```

Comment:
does this have any bearing on the buffers self destructing? is it taking into account how many reference counts there are before destructing?

Reply:
Yes - we don't want to hold on to any other references to the buffers, we want the Table to be the only owner. I'll clarify this part here.